### ISSUE TITLE
`dev-mode` file watching doesn't work on non-root volumes; expose mutagen configr

### ISSUE THREAD

**Issue by philjb** [garden-io/garden#2495]
### Current Behavior

On OSX (at least), Garden [`dev-mode` with Mutagen backend](https://garden.io/blog/kubernetes-hot-reload-revamped-inner-loop-dev-experience/) does not detect file system changes on non-root volumes. I suspect this is a potential general problem not just with OSX with various external and non-root volumes in osx and linux. 

### Expected behavior

Garden `dev-mode` should detect changes for code no matter where it is in the file system (any volume, mount, path). 

### Reproducible example

* On OSX, create a new partition and mount it on the file system
* Clone your code to a new directory in the mounted partition.
* Set up garden dev-mode per the getting started guide.
* Notice that code changes are not detected.

### Workaround

None

(Move code to the root volume but this isn't always any option, e.g. for corporate, security, or technical reasons)

### Suggested solution(s)

Mutagen already provides configuration options for this situation; namely `force-poll` that won't use the native file system watching: https://mutagen.io/documentation/synchronization/watching

Exposing this mutagen option in garden's dev-mode will allow working around the problem.

I believe because Garden generates the mutagen configuration on the fly in a temporary file location that a typescript change needs to be made to allow end users of `dev-mode` to adjust mutagen configuration. I think it'd be nice to accept a separate yml that gets integrated into the auto-generated mutagen configuration so mutagen can be customized in dev-mode.

https://github.com/garden-io/garden/blob/d2828370c439a3053ffaece075b2365e9f9d8fe4/core/src/plugins/kubernetes/dev-mode.ts#L219-L221

to 

```
sync:
  defaults:
    watch:
      mode: "force-poll"
```

or 

allowing cli option `--watch-mode=force-poll` to be passed to Mutagen would work as well.

### Your environment

```
ProductName:    Mac OS X
ProductVersion: 10.15.7
BuildVersion:   19H1217
```

```
gke
```

```
> garden version
0.12.24
```

**Comment by edvald** [comment]
Hey @philjb. I think exposing this option makes perfect sense, I'll look into it as soon as #2491 is merged.

**Comment by philjb** [comment]
@edvald - Since I opened this issue, I found there is a way to specify mutagen preferred defaults with a `~/.mutagen.yml` file. I tried with these contents
```
sync:
  defaults:
    watch:
      mode: force-poll
```

but I had trouble verifying it the yml changed anything. In any case, I think it would be nice to allow customizing how garden runs mutagen in dev-mode in a general way. Thanks for considering it!
